blocks/extensions/saveload.py

```diff
@@ -85,83 +88,78 @@ def do(self, callback_name, *args):
         construction stage.
 
         """
-        from_main_loop, from_user = self.parse_args(callback_name, args)
+        _, from_user = self.parse_args(callback_name, args)
         try:
             path = self.path
             if from_user:
                 path, = from_user
-            already_saved_to = self.main_loop.log.current_row.get(SAVED_TO, ())
-            self.main_loop.log.current_row[SAVED_TO] = (
-                already_saved_to + (path,))
-            secure_pickle_dump(self.main_loop, path)
+            secure_dump(self.main_loop, path, use_cpickle=self.use_cpickle)
             filenames = self.save_separately_filenames(path)
             for attribute in self.save_separately:
-                secure_pickle_dump(getattr(self.main_loop, attribute),
-                                   filenames[attribute])
+                secure_dump(getattr(self.main_loop, attribute),
+                            filenames[attribute], cPickle.dump)
         except Exception:
-            self.main_loop.log.current_row[SAVED_TO] = None
+            path = None
             raise
+        finally:
+            already_saved_to = self.main_loop.log.current_row.get(SAVED_TO, ())
+            self.main_loop.log.current_row[SAVED_TO] = (already_saved_to +
```

Comment:
Yes, that's on purpose. If multiple saves are performed in one iteration, it's very feasible that one failed, but not the other, so why should it be `None` instead of `(None, '/succesful/model')`?